### PR TITLE
[Review - 3] Directory Manager Documentation

### PR DESCRIPTION
**Pull request by satviksr**
Rendered documentation can be seen at https://github.com/googleinterns/filesystem-in-appsscript/blob/docs/DirectoryAPI/docs/DirectoryManager.md

### CODE REVIEW

File: docs/DirectoryManager.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
+# Directory Manager
+
+This module is responsible for managing the file-system state and also provides some low level APIs for managing managed files.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: What are managed files?

File: docs/DirectoryManager.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+> `ActiveWorkbook.path` is the parent directory of the current active workbook
+
+**Or set the current working directory to `ActiveWorkbook.Path`:**
+
+`ChDir (ActiveWorkbook.path)`
+
+> Note: This usage indicates that developers are not using the functionality of `CurDir()` directly. And t
+> hey are not using the persistence ability of the `CurDir()` API.
+
+> `CurDir("D")` returns the current working directory on `D Drive`. The current working drive itself can be changed by `ChDrive("D")`.
```
Comment (monikadhok) [inline docs/DirectoryManager.md:126]: Are you referring to apps script here? Mention so, it is confusing.

File: docs/DirectoryManager.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+### File Path
+
+File paths on windows start with a drive letter followed by a colon. The file separator is a backward slash. File and folde names are **case sensitive**.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: folder*

File: docs/DirectoryManager.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+`C:\Users\Satvik\Desktop\Google`
+
+File paths on macOS doesn't have a drive letter. The file separator is a forward slash. File and folde names are **case insensitive**.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: folder*

File: docs/DirectoryManager.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+`/Users/Satvik/Desktop/Google`
```
Comment (monikadhok) [inline docs/DirectoryManager.md:74]: Mention that `/Users/Satvik/Desktop/Google` and `/Users/satvik/Desktop/Google` mean the same?

File: docs/DirectoryManager.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+VBA also support legacy Macintosh file paths. Apple has deprecated this type of file paths.
+
+`Macintosh HD:Users:Satvik:Desktop:Google`
+
+> For the most part, the main differences between mac and windows file paths are the drive letter and the file separators.
+
+### `ChDir(path)`
+
+The api is used to change the current working directory. The path could either be absolute path or relative path to current working directory.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: to 'the' current..

File: docs/DirectoryManager.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/curdir-function
+
+**VBA API Usage:**
+
+`CurDir [ (drive) ]`
+
+Design:
+
+1. `CurDir` is set to `ActiveWorkbook.path` in the beginning of each macro call. This behaviour differs from Windows but this has been decided as this is the most expected usage.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: This is confusing.In the first point, you mention 
"`CurDir` is set to `ActiveWorkbook.path` in the beginning of each macro call. This behaviour differs from Windows but this has been decided as this is the most expected usage. " what equivalent apps script api is going to do.. But the header is 'VBA API usage' Change the header

File: docs/DirectoryManager.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+**VBA API Usage:**
+
+`CurDir [ (drive) ]`
+
+Design:
+
+1. `CurDir` is set to `ActiveWorkbook.path` in the beginning of each macro call. This behaviour differs from Windows but this has been decided as this is the most expected usage. 
+1. Changes to CurDir does not persist across calls
+1. If a Drive is passed to `CurDir` then, if the drive is same as current working directory, then just return it. If the drive passed is different, then just return root directory. For example `CurDir("E")`, return `E:\`
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: At what point will you resolve "E" using file mapper?

File: docs/DirectoryManager.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+| Part | Description | Implemented | Comment |
+|-|-|-|-|
+| pathname | Optional. String expression that specifies a file name; may include directory or folder, and drive. A zero-length string ("") is returned if pathname is not found. | Yes | Both relative and absolute path work |
+| attributes | Optional. Constant or numeric expression, whose sum specifies file attributes. If omitted, returns files that match pathname but have no attributes. | No | Cannot implement as it is windows file system specific |
+
+Design:
+
+1. Both relative and absolute paths work.
+1. Dir can also take in wild cards. "?" for single character and "*" for multiple characters.
+1. Dir() without a parameter works like a generator, returns the next available match.
```
Comment (monikadhok) [inline docs/DirectoryManager.md]: What is next available match? Ambiguous.

**Review by monikadhok — CHANGES_REQUESTED** [review]

File: docs/DirectoryManager.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+1. Both relative and absolute paths work.
+1. Dir can also take in wild cards. "?" for single character and "*" for multiple characters.
+1. Dir() without a parameter works like a generator, returns the next available match. 
+1. If no match is available, it returns empty string.
+
+> The FileMapper should return the array of all files that match the given pattern
+
+
+
```
Comment (monikadhok) [inline docs/DirectoryManager.md:225]: Can you also add Kill? Seems to have pattern matching challenges there as well.

**Review by monikadhok — CHANGES_REQUESTED** [review]

File: docs/DirectoryManager.md (hunk 10 above)
Comment (monikadhok) [reply docs/DirectoryManager.md:225]: Add a todo here on Muskan for the dependency part. Will be easy to track.

File: docs/DirectoryManager.md (hunk 1 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Fixed, it was a typo

File: docs/DirectoryManager.md (hunk 2 above)
Comment (satviksr) [reply docs/DirectoryManager.md:126]: I rearranged it, so it should be clear now.

File: docs/DirectoryManager.md (hunk 3 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Fixed

File: docs/DirectoryManager.md (hunk 4 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Fixed

File: docs/DirectoryManager.md (hunk 6 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Fixed

File: docs/DirectoryManager.md (hunk 7 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Rearranged, it should be clear now

File: docs/DirectoryManager.md (hunk 8 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: Fixed, I added the point

File: docs/DirectoryManager.md (hunk 9 above)
Comment (satviksr) [reply docs/DirectoryManager.md]: I added an example, should be clear now

File: docs/DirectoryManager.md (hunk 10 above)
Comment (satviksr) [reply docs/DirectoryManager.md:225]: Fixed

File: docs/DirectoryManager.md (hunk 5 above)
Comment (satviksr) [reply docs/DirectoryManager.md:74]: I changed it, so it shouldn't be required anymore

**Review by monikadhok — APPROVED** [review]

**Pull request merged**